lua/entities/gmod_wire_adv_microphone.lua

```diff
@@ -0,0 +1,206 @@
+AddCSLuaFile()
+
+local MIN_VOLUME = 0.02
+local MAX_DIST_GAIN = 1000
+
+local PLAYER_VOICE_MAXDIST_SQR = 250*250
+
+ENT.Type = "anim"
+ENT.Base = "base_wire_entity"
+ENT.Author = "stpM64"
+ENT.PrintName = "Wire Advanced Microphone"
+ENT.Purpose = "Listens to sounds, soundscapes and player voices"
+-- Named 'advanced' because 'gmod_wire_microphone' exists in Wire Extras
+ENT.WireDebugName = "Advanced Microphone"
+
+-- Note: we listen both serverside and clientside,
+-- because some sounds are played clientside only
+
+-- array(Entity(gmod_wire_adv_microphone))
+-- Array instead of lookup table because sounds are emitted more often than microphones switched on or off,
+-- so iteration is more frequent than insertion/removal.
+-- Microphone is live when it is active and at least one active speaker connected to it. 
+_WireLiveMicrophones = _WireLiveMicrophones or {}
+local LiveMics = _WireLiveMicrophones
+
+function ENT:SetupDataTables()
+    self:NetworkVar("Bool", 0, "Active")
+    self:NetworkVarNotify("Active",self.OnActiveChanged)
+end
+
+function ENT:Initialize()
+    if SERVER then
+        self:PhysicsInit( SOLID_VPHYSICS )
+        self:SetMoveType( MOVETYPE_VPHYSICS )
+        self:SetSolid( SOLID_VPHYSICS )
+
+        self.Inputs = WireLib.CreateInputs(self, {
+            "Active"
+        })
+    end
+
+    -- table(Entity(gmod_wire_adv_speaker), true)
+    self._activeSpeakers = {}
+
+    -- Callback not called if 'Active' changes right after creation.
+    self:OnActiveChanged(nil, nil, self:GetActive())
+end
+
+local function Mic_SetLive(self, isLive)
+    if not IsValid(self) then
+        isLive = false
+    else    
+        if self:GetLive() == isLive then return end
+        self:AddEFlags(EFL_FORCE_CHECK_TRANSMIT)
+    end
+
+    if isLive then
+        table.insert(LiveMics, self)
+    else
+        table.RemoveByValue(LiveMics, self)
+    end
+end
+
+ENT.SetLive = Mic_SetLive
+
+function ENT:UpdateTransmitState()
+    return Either(self:GetLive(), TRANSMIT_ALWAYS, TRANSMIT_PVS)
```

Comment:
`Either` only needs to be used in extremely rare cases where the first expression is falsy, which is not true here (even then you could just flip the logic, so this function is kinda redundant)

So just change this to the more common
```lua
self:GetLive() and TRANSMIT_ALWAYS or TRANSMIT_PVS
```